### PR TITLE
charts/portal - Changes to support API Portal SaaS 

### PR DESCRIPTION
**Pull request by hydak**
**Description of the change**

Changed some items which were required as we moved to SaaS.

- parameterized the helpPage to enable externalizing this value
- parameterized the Java Options for 5 services
- added undocumented feature to enable our SaaS rollout to add custom routes
- added undocumented feature to enable an additional port for PortalEnterprise metrics

**Benefits**

This is primary to support our SaaS deployment.  With these changes, our SaaS helm charts and our publicly distributed helm charts are the same, which is better for everybody.

**Drawbacks**

Anywhere I changed, I added the old default values to prevent this change from impacting customers, unless they changed the helms charts themselves.  There are some undocumented options which we are using for SaaS and should consider for public consumption in the future.

**Applicable issues**
None

**Additional information**

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[charts/gateway]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

**Review by APIM-KiranVaddadi — COMMENTED** [review]
By setting  javaOptions -Xms, it becomes minimum resource request mem for the pod (should also add on the os requirements). As Xms and Xmx are same cant we remove the resource limit?.

File: charts/portal/templates/portal-enterprise/portal-enterprise-deployment.yaml [hunk 1]
```diff
@@ -137,11 +137,17 @@ spec:
           {{- if .Values.portalEnterprise.resources }}
           resources: {{- toYaml .Values.portalEnterprise.resources | nindent 12 }}
           {{- end }}
+          {{- if .Values.global.enableMetricEndpoints }}
```
Comment (APIM-KiranVaddadi) [inline charts/portal/templates/portal-enterprise/portal-enterprise-deployment.yaml]: Can be change to 
ports:
   - containerPort: 8080
 {{- if .Values.global.enableMetricEndpoints }}
    - containerPort: 9999
  {{ end }}

Comment (hydak) [reply charts/portal/templates/portal-enterprise/portal-enterprise-deployment.yaml]: I just made a change.  Please validate @APIM-KiranVaddadi

**Comment by hydak** [comment]
@APIM-KiranVaddadi as per your JavaOptions comment, I didn't change the default values.  That's what they were before this PR.  I just parameterized them so they could be overwritten in the values.yaml file.

**Review by APIM-KiranVaddadi — APPROVED** [review]
LGTM

**Pull request merged**